### PR TITLE
if else changed with switch case

### PR DESCRIPTION
**Pull request by MrSagarBiswas**
**What kind of change does this PR introduce?**

if-else statement changed with a switch case statement.

**Did you add tests for your changes?**

No

**Summary**

if-else better for boolean values: If-else conditional branches are great for variable conditions that result into a boolean, whereas switch statements are great for string type values.
Speed: Here number of cases is 16 which is why the switch statement will give the fastest speed.
The clarity in readability: A switch looks much cleaner when you have to combine cases.

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by MrSagarBiswas** [comment]
Just Changed the if-else statements with switch case

File: lib/resolvers/post_query/posts.js [hunk 1]
```diff
@@ -5,41 +5,58 @@ module.exports = async (parent, args) => {
...
-      sort = { videoUrl: -1 };
-    } else if (args.orderBy === 'likeCount_ASC') {
-      sort = { likeCount: 1 };
-    } else if (args.orderBy === 'likeCount_DESC') {
-      sort = { likeCount: -1 };
-    } else if (args.orderBy === 'commentCount_ASC') {
-      sort = { commentCount: 1 };
-    } else {
-      sort = { commentCount: -1 };
+    if (isSortingExecuted) {
```
Comment (JamaicanFriedChicken) [inline lib/resolvers/post_query/posts.js:8]: this is a bit over indented, this is the reason why the linting check is failing

Comment (DMills27) [reply lib/resolvers/post_query/posts.js:8]: Make the changes and we can approve this PR.

Comment (MrSagarBiswas) [reply lib/resolvers/post_query/posts.js]: Done.

**Comment by MrSagarBiswas** [comment]
**What kind of change does this PR introduce?**


if-else statement changed with a switch case statement.

**Did you add tests for your changes?**


No

**Summary**



if-else better for boolean values: If-else conditional branches are great for variable conditions that result into a boolean, whereas switch statements are great for string type values.
Speed: Here number of cases is 16 which is why the switch statement will give the fastest speed.
The clarity in readability: A switch looks much cleaner when you have to combine cases.

**Does this PR introduce a breaking change?**



**Other information**



**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**


Yes

File: lib/resolvers/post_query/posts.js [hunk 2]
```diff
@@ -6,40 +6,57 @@ module.exports = async (parent, args) => {
...
+        sort = { videoUrl: -1 };
+        break;
+      case "likeCount_ASC":
+        sort = { likeCount: 1 };
+        break;
+      case "likeCount_DESC":
+        sort = { likeCount: -1 };
+        break;
+      case "commentCount_ASC":
+        sort = { commentCount: -1 };
```
Comment (JamaicanFriedChicken) [inline lib/resolvers/post_query/posts.js:53]: this should be `1` since its ascending.

**Review by JamaicanFriedChicken — CHANGES_REQUESTED** [review]
found one more issue, just need to fix this then i'll approve it.

**Comment by palisadoes** [comment]
@MrSagarBiswas Please follow the procedure in our CONTRIBUTING.md file. All PRs need to have issues and an assigned person to fix it before the PR is submitted.

We have had a history of unmanageable PRs and this process was introduced to eliminate that risk.

When creating the issue, please justify why you feel changing if-then-else to a case statement is a worthwhile contribution to the repository.

This PR will be closed. Please follow the process.

**Comment by palisadoes** [comment]
We are also focusing on getting our test coverage higher.

If you plan to resubmit the PR, please provide statistics on how much of an improvement in performance you expect, the general industry view on if-then-else versus case statements and include in the PR all instances in the repo that meet this criteria for the overall benefit of the community.

**Pull request closed without merging**